### DISCUSSION TITLE
$ operator is invalid for atomic vectors

### DISCUSSION THREAD

**Discussion by doroft** [LAPKB/Pmetrics#93] [Q&A] [answered]
> exRes$data$plot()
Error in exRes$data : $ operator is invalid for atomic vectors

Hi there, 
This operation runs absolutely fine in the example tutorial data set. In my relatively large data set (~130patients) that has been imperfectly modeled it throws this error when trying to view objects.  Any operation with the data throws the same error. Using the S3 way seems to work i can get summaries and data plotted. It seems to be an issue with a large data set. Any ideas?

**Comment by Siel** [comment]
Hi @doroft, thank you for your report.

Could you post the output of 
`class(exRes)`
and
`exRes`

Thank you!

**Comment by doroft** [comment]
library(Pmetrics)
> 
> setwd("f:/vanc/")
> exData <- PM_data$new(data = "PIMSonly.csv")
DATA STANDARDIZATION REPORT:

 Data are in full format already.

DATA VALIDATION REPORT:

No data errors found.

FIX DATA REPORT:

There were no errors to fix in you data file.
> # MODEL OBJECT
> mod <- PM_model$new("PIMV2test.txt")
> exFit <- PM_fit$new(model = mod, data = exData)
> ##The run
> dir.create("f:/vanc/Runs_Test")
> ##The run
> #dir.create("f:/vanc/Runs_Test")
> setwd("f:/vanc/Runs_Test")
> exFit$run(engine="NPAG",cycles=2000)  ###"runs the run
Creating model file at: genmodel.txt

Model solver mode: ODE
Number of compartments: 2
Primary Variables: V, Cl, IC2, KCP, KPC
Covariates in data file:  ic, age, wt, pelod, vis, scr, alb
Covariates used in model file:  IC, AGE, WT, PELOD, VIS, SCR, ALB
Secondary Variables: KCPW, KPCW, CLW, VW, Ke
Model conditions: no bioavailability term defined, initial conditions are not zero, no lag term defined
Number of cycles to run: 2000

' NPAG  run initiated at 14:44 on 2022 Sep 03. 
[1] "f:/vanc/Runs_Test/1/outputs"
> list.files()
[1] "1"
> ##Post Run analysis
> setwd("f:/vanc/Runs_Test")
> exRes <- PMload(1)
This function is for legacy Pmetrics.
Please see documentation for R6 Pmetrics
      and PM_load
> exRes$data$plot()
Error in exRes$data : $ operator is invalid for atomic vectors
> class(exRes)
[1] "logical"
>

Above is the console output. Including the class(exRes)....

**Comment by Siel** [comment, marked as answer]
Hello @doroft ,

from what I can see, the error is that you’re using PMload instead of PM_load. As the warning there suggests, PMload is a function of the legacy Pmetrics and does not generate the R6 object you need to use the $ operator.

please try the same using PM_load and let me know how it goes

**Comment by Siel** [comment]
@doroft 
I'm going to mark this as answered. Please let us know if it is working ok
